TMEDIA 858 header block

JackHowa opened the pull request:
## Description

Update headers to use design system v2

## Jira Ticket

- [TMEDIA-858](https://arcpublishing.atlassian.net/browse/TMEDIA-858)

## Acceptance Criteria

Header block is migrated to the new architecture. 

Should be updated to use <HeadingSection> and <Heading> components to render headline, and should use size dropdown to apply the style variations not change heading level

Example of Block changes [arc-themes-blocks/default.jsx at 8bb717fb2b9211154b04007eafbb50ca6c1fe907 · WPMedia/arc-themes-blocks](https://github.com/WPMedia/arc-themes-blocks/blob/8bb717fb2b9211154b04007eafbb50ca6c1fe907/blocks/header-block/features/header/default.jsx) 

Example of CSS changes - [arc-themes-blocks/header.scss at 8bb717fb2b9211154b04007eafbb50ca6c1fe907 · WPMedia/arc-themes-blocks](https://github.com/WPMedia/arc-themes-blocks/blob/8bb717fb2b9211154b04007eafbb50ca6c1fe907/blocks/header-block/features/header/header.scss)

## Test Steps

1. Checkout this branch `git checkout TMEDIA-858-header-block-refactor`
2. Run `npm run storybook`
3. Compare with option 1 and hierarchy for typography in figma 
![Screen Shot 2022-05-16 at 13 21 29](https://user-images.githubusercontent.com/5950956/168667323-7e352eee-9513-4dd7-85a2-a3d791f0c110.png)

## Effect Of Changes

### Before

- Hard-coded heading hierarchy 
- Styles determined by news theme css (not block styles)

### After

- Uses helper for establishing hierarchy in the headings from arc themes components
- Uses v2 design system responsive styles based on design discussion

## Dependencies or Side Effects

- manifest https://github.com/WPMedia/arc-themes-manifests/pull/53
- feature pack https://github.com/WPMedia/arc-themes-feature-pack/pull/243

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.


# Questions 

-> using heading level from discussion for each level 
![Screen Shot 2022-05-16 at 13 21 29](https://user-images.githubusercontent.com/5950956/168665856-77ae7753-61b3-4774-aad5-a78d1651a0b9.png)

- There's no global font-size for 30px for large size. In the original mocks, they're listed as 30px as well. What should the "large" size be? 32px seems to be what's recommended for enhanced styling docs for reference for default. But the reference there is starting on a different scale and would result in significantly bigger font-size for desktop
<img width="822" alt="Screen Shot 2022-05-13 at 15 13 07" src="https://user-images.githubusercontent.com/5950956/168382775-f493dfe4-2ad1-4095-ab6d-478fad5519a8.png">

<img width="1176" alt="Screen Shot 2022-05-13 at 15 17 57" src="https://user-images.githubusercontent.com/5950956/168383401-f17389e0-268a-4e99-a68f-f8aab0830a5e.png">

JackHowa committed:
feat: Add header block for establishing the sizes and tokens. Basic setup for v2 architecture

JackHowa committed:
test: Update fallback and class name tests for header

JackHowa committed:
refactor: Reduce code duplication. Add block modifier to ensure one of the sizes still used

Commit:
Added large, medium, and small sizes to the header block

Commit:
Merge branch 'TMEDIA-858-header-block-refactor' of github.com:WPMedia/arc-themes-blocks into TMEDIA-858-header-block-refactor

JackHowa committed:
feat: Use heading section to render based on heading level

JackHowa committed:
style: Remove unused variable

JackHowa committed:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-858-header-block-refactor

JackHowa committed:
style: Update newline per stylint complaining

nschubach commented on blocks/header-block/package.json:
```diff
@@ -23,9 +24,7 @@
   },
   "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95"
-}
+}
```
```suggestion
}

```

nschubach replied:
This is just adding a newline at the EOF

nschubach commented on blocks/header-block/features/header/default.test.jsx:
```diff
@@ -47,81 +36,94 @@ describe("The header block", () => {
+	expect(wrapper.text()).toEqual("");
+	const mediumHeader = wrapper.find(".b-header--medium");
+	expect(mediumHeader).toExist();
+});
```
```suggestion
describe("when no custom fields provided", () => {
	it("should fallback to an empty object and render defaults", () => {
		const wrapper = mount(<Header />);
		expect(wrapper.text()).toEqual("");
		const mediumHeader = wrapper.find(".b-header--medium");
		expect(mediumHeader).toExist();
	});
});
```

nschubach replied:
Should have the "it" in a "describe"

JackHowa replied:
updated

nschubach reviewed (dismissed):
Just a few couple changes I see.

nschubach commented on .storybook/themes/news.scss:
```diff
@@ -334,3 +353,4 @@
 
 @use "../../blocks/right-rail-block";
 @use "../../blocks/right-rail-advanced-block";
+@use "../../blocks/header-block";
```
Can we alphabetize this?  The list is probably going to get long and finding things would be easier.

JackHowa replied:
updated

JackHowa committed:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-858-header-block-refactor

JackHowa committed:
feat: Add default and desktop line height and font size.

Commit:
Add default and desktop line height and font size fir large, medium and small sizes

Commit:
Update blocks/header-block/package.json

Co-authored-by: nschubach <nschubach@gmail.com>

Commit:
Update blocks/header-block/features/header/default.test.jsx

Co-authored-by: nschubach <nschubach@gmail.com>

JackHowa committed:
style: Update alphabetization based on feedback

https://github.com/WPMedia/arc-themes-blocks/pull/1344#pullrequestreview-974058985

Commit:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-858-header-block-refactor

JackHowa committed:
styles: Remove extra space. Fix alphabetization

JackHowa committed:
fix: Duplicate headline import on bad merge
